hivemind/optim/collaborative.py

```diff
@@ -236,6 +236,40 @@ def step(self, batch_size: Optional[int] = None, **kwargs):
 
             return group_info
 
+    def step_aux(self, **kwargs):
+        """
+        Find and assist other peers in averaging -- without sending local gradients to those peers.
+
+        :note: this .step is different from normal pytorch optimizers in several key ways. See __init__ for details.
+        """
+
+        if not self.collaboration_state.ready_for_step:
+            return
+
+        logger.log(self.status_loglevel,
+                   f"Beginning global optimizer step {self.collaboration_state.optimizer_step}")
+        self.collaboration_state = self.fetch_collaboration_state()
+        self.collaboration_state_updated.set()
```

Comment:
These lines (and also some below) execute the same code as in `step`, maybe extract them to methods to reduce code duplication?

Reply:
Yes, probably.
May I say for now that duplication is not too big? =)

Reply:
It is, agreed